### DISCUSSION TITLE
See table columns properties ordered by position

### DISCUSSION THREAD

**Discussion by carlosherranz** [dbeaver/dbeaver#16110] [Q&A]
When I use table properties screen panel (F4 on table name), table columns always appears alphabeticaly and I can't find where in preferences I can change this. 
How can I see columns properties of a table ordered by position?

Thanks

**Comment by LonwoLonwo** [comment]
Hello @carlosherranz 

Usually DBeaver shows column correctly by theirs order positions.
So maybe this is a bug.

What Dbeaver version do you use? What Database and driver do you use? Can you provide some screenshot of this behaviour?

![2022-04-07 16_17_18-DBeaver Enterprise 21 3 3 - certification](https://user-images.githubusercontent.com/45152336/162208182-0c549d96-6dcd-49c6-b54c-454024bfe9df.png)

**Comment by carlosherranz** [comment]
Hello Anastasiya

I’m using Version 22.0.2 DBeaver Community Edition.
The database is DB2/iSeries400, the driver is JTopen 10.7, released 15 September 2021.
And the screenshot asked is attached below

I think it can be a preferences configuration problem, but I don’t find what or where I have to change it

![2022-04-07 16_30_15-Clipboard](https://user-images.githubusercontent.com/28946478/162223251-afe54d21-5a3e-4dbd-9060-7c34e90e3d7e.jpg)

**Comment by LonwoLonwo** [comment]
Sorry, you can add screenshots via e-mail. You can add it here in Github.

**Comment by carlosherranz** [comment]
I have updated the response directly on GitHub and I add the screenshot also

**Comment by LonwoLonwo** [comment]
So, I can see that columns have order. 

And I found this setting in Preferences, maybe it is enabled in your settings? (Preferences ->User Interface -> Navigator)

![2022-04-07 17_48_12-Preferences](https://user-images.githubusercontent.com/45152336/162227735-e4e7be9b-f282-402f-a912-691c73a313a4.png)

**Reply by carlosherranz** [reply]
I think I have seen what happens. You're right, checking order elements alphabetically shows columns alphabetically, and unchecking by position. The problem is when you check or uncheck and press F4 on the same table, here doesn't work fine, but if you press on another it does.  
When you check or uncheck with the same table, you have to exit and start DBeaver again and work fine always, or if you don't want to exit, as you say, the remedy is to sort columns manually.

**Reply by LonwoLonwo** [reply]
So... Is it working now?

**Reply by carlosherranz** [reply]
Yes, thank you

**Reply by LonwoLonwo** [reply]
Great! Thanks for the answer.

**Comment by carlosherranz** [comment]
I havet tried that option enable and disable and the results are the same, ordered always alphabeticaly.

Perhaps that option is not for that purpose

**Comment by LonwoLonwo** [comment]
Can you provide your driver settings?

![2022-04-07 18_41_03-DBeaver Enterprise 21 3 3 - EMP](https://user-images.githubusercontent.com/45152336/162239309-478405e9-8e53-4700-ab52-715297d374c9.png)

Also, as a workaround, you can sort columns manually.

![2022-04-07 18_42_25-DBeaver Enterprise 21 3 3 - EMP](https://user-images.githubusercontent.com/45152336/162239851-eec9e80e-9cc2-4d76-bc00-4981fb474c8f.png)
